File: backend/app/use_cases/user/delete_user.py
from app.core.exceptions import BusinessError
from app.domain.repositories.user_repository import IUserRepository


class DeleteUser:
    def __init__(self, user_repository: IUserRepository):
        self.user_repository = user_repository

    async def execute(self, user_id: str) -> None:
        # Verificar se o usuário existe
        existing_user = await self.user_repository.find_by_id(user_id)
        if not existing_user:
            raise BusinessError("Usuário não encontrado")

        # Deletar usuário
        await self.user_repository.delete(user_id)

File: frontend/src/core/use-cases/user/DeleteUser.ts
import { UserRepository } from '../../interfaces/repositories/UserRepository';

export class DeleteUser {
  constructor(private userRepository: UserRepository) {}

  async execute(id: string): Promise<void> {
    // Verificar se o usuário existe
    const existingUser = await this.userRepository.findById(id);
    if (!existingUser) {
      throw new Error('Usuário não encontrado');
    }

    // Deletar o usuário
    await this.userRepository.delete(id);
  }
} 